functionary/ui/templatetags/social_helper.py

```diff
@@ -0,0 +1,23 @@
+from allauth.socialaccount.providers.base import ProviderAccount
+from django import template
+
+register = template.Library()
+
+
+@register.simple_tag(takes_context=True)
+def find_account(context, provider_id: str) -> ProviderAccount | None:
+    """This tag checks the SocialAccounts in the form found in the
+    templates context and returns the desired ProviderAccount.
+
+    Args:
+        context: the template's context, automatically passed in
+        provider_id: ID of the SocialAccount to find
+
+    Returns:
+        The matching ProviderAccount or None
+    """
+    if "form" in context:
```

Comment:
Small nitpick: to keep this pythonic, this could just be `if context.get("form")`

Reply:
More specifically, something like:

```python
if form := context.get("form")
    for account in form.accounts:
       # carry on as before
```

Reply:
fixed